Annotation scanner NPE when processing JAX-RS endpoints with params of type SortedSet

Hi, I've noticed that some RESTEasy tests failing against Wildfly when declaring REST resources methods accepting parameters of type `SortedSet`. 
Maybe I am missing something but looks like this is allowed by JAX-RS spec: https://docs.oracle.com/javaee/7/api/javax/ws/rs/FormParam.html 

The following stack trace represents the output of a deployment failure when an attempt to deploy a WAR with such a resource:

```
Caused by: java.lang.NullPointerException
	at io.smallrye.openapi.runtime.scanner.ParameterProcessor.setSchemaProperties(ParameterProcessor.java:637)
	at io.smallrye.openapi.runtime.scanner.ParameterProcessor.getFormBodyContent(ParameterProcessor.java:605)
	at io.smallrye.openapi.runtime.scanner.ParameterProcessor.process(ParameterProcessor.java:404)
	at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.processJaxRsMethod(OpenApiAnnotationScanner.java:577)
	at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.lambda$processJaxRsResourceClass$0(OpenApiAnnotationScanner.java:422)
...
```

I've isolated a reproducer [1] which has three methods to demonstrate the failure: the first and the second ones (namely `formAsListOfString` and `formAsSetOfString`) take a `List` and a `Set` and are working fine, while the last one - `formAsSortedSetOfString` -, taking a `SortedSet`, fails. 
The methods are using `@FormParam` because it was the first case that I spotted but I've seen failures when using the same data type with `@QueryParam` and `@MatrixParam` as well.

The reproducer should fail by default. For a successful deploy the `formAsSortedSetOfString` method should be commented.

[1]
https://github.com/fabiobrz/microprofile-openapi-example/tree/sorted-set-repro

--- a/core/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiDataObjectScanner.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiDataObjectScanner.java
@@ -71,6 +71,9 @@ public class OpenApiDataObjectScanner {
     // Map
     public static final DotName MAP_INTERFACE_NAME = DotName.createSimple(Map.class.getName());
     public static final Type MAP_TYPE = Type.create(MAP_INTERFACE_NAME, Type.Kind.CLASS);
+    // Set
+    public static final DotName SET_INTERFACE_NAME = DotName.createSimple(java.util.Set.class.getName());
+    public static final Type SET_TYPE = Type.create(SET_INTERFACE_NAME, Type.Kind.CLASS);
     // Enum
     public static final DotName ENUM_INTERFACE_NAME = DotName.createSimple(Enum.class.getName());
     public static final Type ENUM_TYPE = Type.create(ENUM_INTERFACE_NAME, Type.Kind.CLASS);

--- a/core/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/TypeProcessor.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/TypeProcessor.java
@@ -6,6 +6,7 @@ import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.ENUM_
 import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.ITERABLE_TYPE;
 import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.MAP_TYPE;
 import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.OBJECT_TYPE;
+import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.SET_TYPE;
 import static io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.STRING_TYPE;
 import static io.smallrye.openapi.runtime.util.TypeUtil.isTerminalType;
 
@@ -159,6 +160,10 @@ public class TypeProcessor {
             DataObjectLogging.log.processingTypeAs("Java Collection", "Array");
             Schema arraySchema = new SchemaImpl();
             schema.type(Schema.SchemaType.ARRAY);
+
+            if (TypeUtil.isA(index, pType, SET_TYPE)) {
+                schema.setUniqueItems(Boolean.TRUE);
+            }
 
             // Should only have one arg for collection.
             Type arg = pType.arguments().get(0);

--- a/core/src/main/java/io/smallrye/openapi/runtime/util/TypeUtil.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/util/TypeUtil.java
@@ -10,9 +10,11 @@ import java.util.Collection;
 import java.util.Collections;
 import java.util.Date;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
@@ -337,7 +339,10 @@ public class TypeUtil {
     }
 
     static ClassInfo getClassInfo(IndexView appIndex, Type type) {
-        DotName className = getName(type);
+        return getClassInfo(appIndex, getName(type));
+    }
+
+    static ClassInfo getClassInfo(IndexView appIndex, DotName className) {
         ClassInfo clazz = appIndex.getClassByName(className);
         if (clazz == null) {
             clazz = jdkIndex.getClassByName(className);
@@ -369,37 +374,39 @@ public class TypeUtil {
         // First, look in Jandex, as target might not be in our classloader
         ClassInfo subJandexKlazz = getClassInfo(index, testSubject);
 
-        if (subJandexKlazz != null) {
-            return subJandexKlazz.interfaceNames().contains(getName(testObject)) || hasSuper(index, subJandexKlazz, testObject);
+        if (subJandexKlazz != null && superTypes(index, subJandexKlazz).contains(getName(testObject))) {
+            return true;
         }
 
         return isAssignableFrom(testSubject.name(), testObject.name());
     }
 
-    private static boolean hasSuper(IndexView index, ClassInfo testSubject, Type testObject) {
-        Type superKlazzType = testSubject.superClassType();
+    private static Set<DotName> superTypes(IndexView index, ClassInfo testSubject) {
+        Set<DotName> superTypes = new HashSet<>();
 
-        while (superKlazzType != null) {
-            if (getName(superKlazzType).equals(getName(testObject))) {
-                return true;
-            }
-            if (DOTNAME_OBJECT.equals(getName(superKlazzType))) {
-                return false;
-            }
+        testSubject.interfaceNames().forEach(iface -> {
+            superTypes.add(iface);
 
-            ClassInfo superKlazz = getClassInfo(index, superKlazzType);
+            ClassInfo superIFace = getClassInfo(index, iface);
+
+            if (superIFace != null) {
+                superTypes.addAll(superTypes(index, superIFace));
+            }
+        });
+
+        Type superType = testSubject.superClassType();
+
+        if (superType != null) {
+            superTypes.add(getName(superType));
+
+            ClassInfo superKlazz = getClassInfo(index, superType);
 
             if (superKlazz != null) {
-                if (superKlazz.interfaceNames().contains(getName(testObject))) {
-                    return true;
-                }
-            } else {
-                return isAssignableFrom(testSubject.name(), testObject.name());
+                superTypes.addAll(superTypes(index, superKlazz));
             }
-
-            superKlazzType = superKlazz.superClassType();
         }
-        return false;
+
+        return superTypes;
     }
 
     public static boolean isTerminalType(Type type) {

--- a/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/ParameterScanTests.java
+++ b/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/ParameterScanTests.java
@@ -320,7 +320,8 @@ public class ParameterScanTests extends IndexScannerTestBase {
         @POST
         @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
         @Produces(MediaType.APPLICATION_JSON)
-        public Widget update(@FormParam("form-param2") @Size(max = 10) String formParam2) {
+        public Widget update(@FormParam("form-param2") @Size(max = 10) String formParam2,
+                @FormParam("qualifiers") java.util.SortedSet<String> qualifiers) {
             return null;
         }
     }

--- a/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/params.path-param-with-form-params.json
+++ b/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/params.path-param-with-form-params.json
@@ -26,6 +26,13 @@
                   "form-param2": {
                     "type": "string",
                     "maxLength": 10
+                  },
+                  "qualifiers": {
+                    "uniqueItems": true,
+                    "type": "array",
+                    "items": {
+                      "type": "string"
+                    }
                   }
                 }
               }

--- a/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.generic-collection.set-indexed.json
+++ b/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.generic-collection.set-indexed.json
@@ -88,6 +88,7 @@
       },
       "SetFruit": {
         "type": "array",
+        "uniqueItems": true,
         "items": {
           "$ref": "#/components/schemas/Fruit"
         }

--- a/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.generic-collection.set-unindexed.json
+++ b/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.generic-collection.set-unindexed.json
@@ -10,6 +10,7 @@
               "application/json": {
                 "schema": {
                   "type": "array",
+                  "uniqueItems": true,
                   "items": {
                     "$ref": "#/components/schemas/Fruit"
                   }
@@ -36,6 +37,7 @@
               "application/json": {
                 "schema": {
                   "type": "array",
+                  "uniqueItems": true,
                   "items": {
                     "$ref": "#/components/schemas/Fruit"
                   }
@@ -62,6 +64,7 @@
               "application/json": {
                 "schema": {
                   "type": "array",
+                  "uniqueItems": true,
                   "items": {
                     "$ref": "#/components/schemas/Fruit"
                   }

